### PR TITLE
Conversion between radians and degrees.Fixes #2881

### LINKED ISSUE #2881: math.nim: trigonometry: radians to degrees conversion
This is a feature request.
 ([relates to Issue 1357](https://github.com/Araq/Nim/issues/1357) )

1.  Could some conversion procs be added to help convert between radians and degrees

```Nim
let  r2d = PI / 180.0

proc degrees2Radians*[int|float](d: T): float {.inline.} = 
  ## Convert degrees to randians, for trigonometric functions
  result = r2d * d.float
  
proc radians2Degrees*[int|float](d: T): float {.inline.} = 
  ## Convert radians to degrees, to make radians more readable
  result = d.float / r2d
```
2. Update the doco to state that trigonometric functions work on radians, not degrees or gradients

### PR DESCRIPTION
**Pull request by apense**
provides a new constant for conversion (not public, but that can be changed if it's wanted). the functions are pretty simple, honestly. top-level comment so documentation for each function could remain. Fixes #2881

### CODE REVIEW

**Comment by yglukhov** [comment]
IMHO, it would be much more convenient if the procs return float64 for float64 arg, and float32 for float32 arg.

**Comment by apense** [comment]
I can do that, but I'm a little hesitant, since that behavior doesn't seem to show up in the rest of [math.nim](https://github.com/nim-lang/Nim/blob/devel/lib/pure/math.nim). 
Although, I should just take out the `T` and let it be float, since the conversion to float from int or whatever will happen naturaly.

**Comment by yglukhov** [comment]
Ahh, that makes sense. But I wouldn't say math.nim does the rightest thing ;). Overloads for both 32 and 64 bits would be really nice, imho. My primary concern is that those functions have unneeded overhead when working with opengl, which operates on 32-bit floats, and opengl is just a tiny example. I agree that the way I want deg-rad conversion to work is inconsistent with current math.nim. Maybe that stuff should be tweaked as well?

File: lib/pure/math.nim [hunk 1]
```diff
@@ -12,6 +12,14 @@
 ## Basic math routines for Nim.
 ## This module is available for the `JavaScript target
 ## <backends.html#the-javascript-target>`_.
+## 
+## Note that the trigonometric functions naturally operate on radians.
+## The helper functions `degToRad` and `radToDeg` provide conversion
+## between radians and degrees.
+## 
+## Note that the trigonometric functions naturally operate on radians.
```
Comment (Araq) [inline lib/pure/math.nim]: Duplicated paragraph.

File: lib/pure/math.nim [hunk 2]
```diff
@@ -317,6 +326,14 @@ else:
 
 {.pop.}
 
+proc degToRad*[T](d: T): float {.inline.} =
```
Comment (Araq) [inline lib/pure/math.nim]: I don't think these should be generic. Also I agree there should be float32/float64 variants instead.

**Comment by apense** [comment]
Are the other math procedures that utilize `float` going to be altered to `float32|float64`? I can go ahead an do that if it's agreed upon. I'm going to write documentation and code examples for them anyways

**Comment by Araq** [comment]
Yeah but ensure it works. For most importc'ed stuff you cannot use float32|float64 but only overloading.

**Comment by apense** [comment]
I'll put all that in a separate PR, then because they're altered procs instead of new

**Pull request merged**